API/Controllers/UserController.cs

```diff
@@ -147,6 +152,7 @@ public async Task<IActionResult> UpdateAccount(int userId, [FromBody] UserResour
         ///     Gets the student information.
         /// </summary>
         /// <returns></returns>
+        [ObsoleteAttribute("This endpoint will soon be deprecated. Use [Post] {userId}/allData.")]
```

Comment:
This obsolete warning can be removed. I'm pretty sure "delete data" is the same as "delete account" right? If so, we should move the logic of the new DeleteAllUserData method into this endpoint and use this one instead.

Reply:
We can indeed change it to the delete endpoint, but for now we added the functionality in a seperate endpoint as we were not sure if the deletion of a user was the same of deleting all data associated with a user.

Reply:
For now I will move the delete function. I will keep the collectAllUserData in a seperate endpoint for future use as a lot of logic now relies on the default GetUser.